### PR TITLE
Adding new MODS fields as per Metadata Interest Group mappings

### PR DESCRIPTION
**Pull request by Natkeeran**
**GitHub Issue**: (link)
* https://github.com/Islandora-CLAW/CLAW/issues/1083

# What does this Pull Request do?
This PR adds various fields needed for a default MODS form as per the above noted ticket.  

# How should this be tested?
* SSH into the vagrant
* Go to Features: http://localhost:8000/admin/config/development/features
* Select Islandora bundle, click newly detected, import changes
* Add a vocabulary list item if needed in the taxonomy
* Create a new repository item
* Verify that new fields appear in the form and display

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora-CLAW/committers

### CODE REVIEW

**Comment by MarcusBarnes** [comment]
I've just tested and this appears to work as expected.

**Comment by whikloj** [comment]
This is failing travis and I'm guessing due to the dependency on the various taxonomy configs which don't exist here. Did you mean to add them @Natkeeran

**Comment by seth-shaw-unlv** [comment]
My guess is that the updates to controlled access terms needs to merge first.

**Comment by whikloj** [comment]
Relies on https://github.com/Islandora-CLAW/controlled_access_terms/pull/35

**Comment by rosiel** [comment]
@Natkeeran the merge conflicts are, as far as i can tell, due to the addition of the weight field at the spots where your changes began. They're not too hard to resolve.

File: config/install/field.field.node.islandora_object.field_coordinates.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+entity_type: node
+bundle: islandora_object
+label: Coordinates
+description: ''
+required: false
+translatable: false
+default_value: {  }
+default_value_callback: ''
+settings: {  }
+field_type: geolocation
```
[the remark this answers is not in the document]
Comment (seth-shaw-unlv) [reply config/install/field.field.node.islandora_object.field_coordinates.yml]: The field comes from https://www.drupal.org/project/geolocation; although I am open to other options.

File: config/install/field.field.node.islandora_object.field_dewey_classification.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+langcode: en
+status: true
+dependencies:
+  config:
+    - field.storage.node.field_dewey_classification
+    - node.type.islandora_object
+id: node.islandora_object.field_dewey_classification
+field_name: field_dewey_classification
```
[the remark this answers is not in the document]
Comment (Natkeeran) [reply config/install/field.field.node.islandora_object.field_dewey_classification.yml]: @rosiel 
Can easily add this.  These fields as as they exist in 7.x book content model.

**Comment by Natkeeran** [comment]
@rosiel thank you for the feedback.

Just getting to this one.  I will make the requested changes.

File: config/install/field.field.node.islandora_object.field_coordinates.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Natkeeran) [reply config/install/field.field.node.islandora_object.field_coordinates.yml]: I am open to suggestions.  Please advise.

[the remark this answers is not in the document]
Comment (rosiel) [reply config/install/field.field.node.islandora_object.field_coordinates.yml]: The 7.x Islandora Image form includes a single text field for "Coordinates". (I assume this is the same for other fields that use "coordinates"). 

cartographics

According to [this geolocation-migration-related issue](https://www.drupal.org/project/geolocation/issues/2776595), a geolocation field can have values migrated into it, by assigning values to `'field_coordinates/lat'` and `'field_coordinates/lng'`. We will need to find or write an algorithm to transform the many ways someone could have written coordinates in their 7.x form (uncontrolled string) into double-precision numbers ("decimal degrees") as expected by geolocation. 

Here are some valid coordinate expressions, [according to ARCGIS](https://pro.arcgis.com/en/pro-app/tool-reference/data-management/supported-notation-formats.htm#ESRI_SECTION1_06E59198BAE24A72BD4E3CE48B99224A) (these are just the ones using degree-based formats):

`27.00N 087.00W`
`27.00n 087.00w`
`27.00N 087W`
`+27.00 087W`
`27N -87N`
`27.00 W087`
`27.00N/87W`
`27.00N/87W`
`27.00°N 087.00°W`

`27 54.00N 087 59.00W`
`27 54.00n 087 59.00w`
`27 54N 087 0W+`
`27 54.00 087 59.00W`
`N27 54.00 W087 59.00`
`27 54.00N/87 59W`
`27° 54.00’N 087° 59.00’W`

`27 18 00.00N 087 00 00.00W`
`271800.00N 0870000.00W`

And it's not listed here but there are also commas I'm pretty sure I've seen comma-separated formats, as in `15.23456, -30.67890`.

We have not in the past required any consistent way of representing coordinates. What is being proposed seems innocuous enough - it's a coordinate field - but it severely impacts the allowed values. I'm on the side of "don't corrupt metadata values in the migration process" over "adding new map visualization features" by default. 

As I'm reading more, I'm in general becoming more uncomfortable with transforming metadata as an automatic process that the user doesn't have a chance to verify before data gets lost.

**Comment by Natkeeran** [comment]
@rosiel 

I've made all the requested changes.  

The vocabs have been made translatable, thus whether field is transtlable or not does not seem matter.

**Comment by Natkeeran** [comment]
Hoping this gets reviewed before the next release!

**Comment by seth-shaw-unlv** [comment]
@Natkeeran, we need updates to https://github.com/Islandora/controlled_access_terms/pull/35 approved and merged before we can do this one.

**Comment by seth-shaw-unlv** [comment]
I know there is a big thread of discussion about what needs to be adjusted. Let me see if I can go through and summarize them.

**Comment by Natkeeran** [comment]
@seth-shaw-unlv oh, right.  Thank you.

**Comment by Natkeeran** [comment]
@dannylamb @rosiel 

I've made the changes to controlled_access_terms as per Seth's suggestions.  Mainly, moving the country vocab and fields into a sub module.  

The only thing that I am considering adding would be add a hook to create an rdf entry for place_country field when the module gets installed.  However, that can be a separate PR.

**Comment by MarcusBarnes** [comment]
I've just tested this PR and it appears to be functioning as expected.

File: config/install/field.field.node.islandora_object.field_table_of_contents.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+entity_type: node
+bundle: islandora_object
+label: 'Table Of Contents'
+description: ''
+required: false
+translatable: true
+default_value: {  }
+default_value_callback: ''
+settings: {  }
+field_type: string_long
```
Comment (seth-shaw-unlv) [inline config/install/field.field.node.islandora_object.field_table_of_contents.yml]: As per [the Slack discussion](https://islandora.slack.com/archives/CM9CVQWJ0/p1579101817013800), please change to `text_long`.

File: config/install/field.storage.node.field_note.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  module:
+    - field_permissions
+    - node
+third_party_settings:
+  field_permissions:
+    permission_type: public
+id: node.field_note
+field_name: field_note
+entity_type: node
+type: string_long
```
Comment (seth-shaw-unlv) [inline config/install/field.storage.node.field_note.yml]: Use `text_long`

File: config/install/field.field.node.islandora_object.field_note.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+entity_type: node
+bundle: islandora_object
+label: Note
+description: ''
+required: false
+translatable: true
+default_value: {  }
+default_value_callback: ''
+settings: {  }
+field_type: string_long
```
Comment (seth-shaw-unlv) [inline config/install/field.field.node.islandora_object.field_note.yml]: Use `text_long`

File: config/install/field.storage.node.field_table_of_contents.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  module:
+    - field_permissions
+    - node
+third_party_settings:
+  field_permissions:
+    permission_type: public
+id: node.field_table_of_contents
+field_name: field_table_of_contents
+entity_type: node
+type: string_long
```
Comment (seth-shaw-unlv) [inline config/install/field.storage.node.field_table_of_contents.yml]: Use `text_long`

File: config/install/core.entity_form_display.node.islandora_object.default.yml [hunk 7]
```diff
@@ -178,8 +313,32 @@ content:
     region: content
     settings: {  }
     third_party_settings: {  }
+  field_note:
+    weight: 30
+    settings:
+      rows: 5
+      placeholder: ''
+    third_party_settings: {  }
+    type: string_textarea
```
Comment (seth-shaw-unlv) [inline config/install/core.entity_form_display.node.islandora_object.default.yml]: Use `text_default`

File: config/install/core.entity_form_display.node.islandora_object.default.yml [hunk 8]
```diff
@@ -194,14 +353,40 @@ content:
...
+    third_party_settings: {  }
+    type: entity_reference_autocomplete
+    region: content
+  field_table_of_contents:
+    weight: 14
+    settings:
+      rows: 5
+      placeholder: ''
+    third_party_settings: {  }
+    type: string_textarea
```
Comment (seth-shaw-unlv) [inline config/install/core.entity_form_display.node.islandora_object.default.yml]: Use `text_default`.

File: config/install/core.entity_form_display.node.islandora_object.default.yml (hunk 7 above)
Comment (seth-shaw-unlv) [reply config/install/core.entity_form_display.node.islandora_object.default.yml]: Correction, widgets use `text_textarea`.

Comment (seth-shaw-unlv) [inline config/install/core.entity_form_display.node.islandora_object.default.yml]: Use `text_textarea`.

File: config/install/core.entity_view_display.node.islandora_object.binary.yml [hunk 9]
```diff
@@ -130,46 +242,78 @@ content:
...
     label: above
     settings:
-      link_to_entity: false
-      native_language: false
+      link: true
     third_party_settings: {  }
-  links:
-    weight: 2
+  field_table_of_contents:
+    type: basic_string
```
Comment (seth-shaw-unlv) [inline config/install/core.entity_view_display.node.islandora_object.binary.yml]: Use `text_default`.

File: config/install/core.entity_view_display.node.islandora_object.default.yml [hunk 10]
```diff
@@ -131,29 +268,43 @@ content:
...
+      link: true
+    third_party_settings: {  }
+    type: entity_reference_label
+    region: content
+  field_table_of_contents:
+    weight: 16
+    label: above
+    settings: {  }
+    third_party_settings: {  }
+    type: basic_string
```
Comment (seth-shaw-unlv) [inline config/install/core.entity_view_display.node.islandora_object.default.yml]: Use `text_default`.

File: config/install/core.entity_view_display.node.islandora_object.open_seadragon.yml [hunk 11]
```diff
@@ -125,38 +237,75 @@ content:
...
+    weight: 25
     region: content
     label: above
     settings:
-      link_to_entity: false
-      native_language: false
+      link: true
+    third_party_settings: {  }
+  field_table_of_contents:
+    type: basic_string
```
Comment (seth-shaw-unlv) [inline config/install/core.entity_view_display.node.islandora_object.open_seadragon.yml]: Use `text_default`.

File: config/install/core.entity_view_display.node.islandora_object.pdfjs.yml [hunk 12]
```diff
@@ -135,28 +265,42 @@ content:
...
+    weight: 25
     region: content
     label: above
     settings:
-      link_to_entity: false
-      native_language: false
+      link: true
+    third_party_settings: {  }
+  field_table_of_contents:
+    type: basic_string
```
Comment (seth-shaw-unlv) [inline config/install/core.entity_view_display.node.islandora_object.pdfjs.yml]: Use `text_default`.

File: config/install/core.entity_view_display.node.islandora_object.binary.yml [hunk 13]
```diff
@@ -130,46 +242,78 @@ content:
     third_party_settings: {  }
     type: entity_reference_label
     region: content
+  field_note:
+    type: basic_string
```
Comment (seth-shaw-unlv) [inline config/install/core.entity_view_display.node.islandora_object.binary.yml]: Use `text_default`.

File: config/install/core.entity_view_display.node.islandora_object.default.yml [hunk 14]
```diff
@@ -121,8 +235,31 @@ content:
     settings:
       link: true
     third_party_settings: {  }
+  field_note:
+    weight: 32
+    label: above
+    settings: {  }
+    third_party_settings: {  }
+    type: basic_string
```
Comment (seth-shaw-unlv) [inline config/install/core.entity_view_display.node.islandora_object.default.yml]: Use `text_default`.

File: config/install/core.entity_view_display.node.islandora_object.open_seadragon.yml [hunk 15]
```diff
@@ -125,38 +237,75 @@ content:
     third_party_settings: {  }
     type: entity_reference_label
     region: content
+  field_note:
+    type: basic_string
```
Comment (seth-shaw-unlv) [inline config/install/core.entity_view_display.node.islandora_object.open_seadragon.yml]: Use `text_default`.

File: config/install/core.entity_view_display.node.islandora_object.pdfjs.yml [hunk 16]
```diff
@@ -117,16 +232,31 @@ content:
     settings:
       link: true
     third_party_settings: {  }
-  field_pid:
-    weight: 15
+  field_note:
+    type: basic_string
```
Comment (seth-shaw-unlv) [inline config/install/core.entity_view_display.node.islandora_object.pdfjs.yml]: Use `text_default`.

**Pull request merged**